custom_hooks/useOnWindowResize.js

```diff
@@ -0,0 +1,33 @@
+import { useState, useEffect } from "react";
+
+function debounce(func, milliseconds) {
+    let timer
+    return () => {
+        clearTimeout(timer)
+        timer = setTimeout(() => {
+            timer = null
+            func.apply(this, arguments)
+        }, milliseconds)
+    };
+}
+
+export const useOnWindowResize = () => {
+    const [dimensions, setDimensions] = useState({ 
+        height: window.innerHeight,
+        width: window.innerWidth
+    })
+
+    useEffect(() => {  
+        const debounceHandleResize = debounce(() => {
+            setDimensions({
+                height: window.innerHeight,
+                width: window.innerWidth
+            })
+        }, 250)
+        window.addEventListener('resize', debounceHandleResize);
+        return () => {
+            window.removeEventListener('resize', debounceHandleResize);
+        }
+    })
```

Comment:
doesn't this need dependency array?
its so tricky IMO

Reply:
slider has loop in current api.deriv.com website
is it possible for you to make the slides loop?

Reply:
Making the slides loop was a challenge, but I managed to implement it now :D